src/components/ApiDocs/ApiModalProvider.tsx

```diff
@@ -1,10 +1,21 @@
-import { useState, createContext } from 'react';
+import { useState, createContext, useRef, RefObject } from 'react';
 import { LinkDataType } from './display/TypeLink';
 import { ApiModal } from './display/ApiModal';
 
-export const TypeContext = createContext({
+interface TypeContextInterface {
+  setModalData: (data: any) => void;
+  setModalTriggerRef: (ref: RefObject<HTMLButtonElement> | null) => void;
+  modalTriggerRef: RefObject<HTMLButtonElement> | null;
+  openModal: () => void;
+  addBreadCrumb: (data: any) => void;
+  setBC: (data: any) => void;
+}
+
+export const TypeContext = createContext<TypeContextInterface>({
   setModalData: (data) => data,
-  modalOpen: () => {},
+  setModalTriggerRef: (ref) => ref,
+  modalTriggerRef: null,
+  openModal: () => {},
```

Comment:
I renamed `modalOpen` -> `openModal` so it's a little more action-sounding, otherwise my first thought was that it is a boolean.